modules/tide_landing_page/src/Plugin/CornerGraphicField.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Drupal\tide_landing_page\Plugin;
+
+use Drupal\Core\Field\FieldItemList;
+use Drupal\Core\TypedData\ComputedItemListTrait;
+use Drupal\node\NodeInterface;
+use Drupal\taxonomy\Entity\Term;
+
+/**
+ * Corner Graphic field implementation.
+ */
+class CornerGraphicField extends FieldItemList {
+
+  use ComputedItemListTrait;
+
+  /**
+   * Computes the field value.
+   */
+  protected function computeValue() {
+    $node = $this->getEntity();
+    if (!$node instanceof NodeInterface) {
+      return;
+    }
+
+    $site_id = \Drupal::request()->get('site');
+    $siteHelper = \Drupal::service('tide_site.helper');
+    $file_url_generator = \Drupal::service('file_url_generator');
+
+    $sites = $siteHelper->getEntitySites($node);
+    $valid = $siteHelper->isEntityBelongToSite($node, $site_id);
+
+    if (!$valid || empty($sites['sections'][$site_id])) {
+      return;
+    }
+
+    $section_id = $sites['sections'][$site_id];
+    $details = ['site_section_id' => $section_id];
+
+    // Try to get corner graphics from the node.
+    if ($this->hasCornerGraphicsInNode($node)) {
+      $this->addNodeCornerGraphics($node, $details, $file_url_generator);
+    }
+    // Fall back to section term corner graphics.
+    else {
+      $term = Term::load($section_id);
+      if ($term && $this->hasCornerGraphicsInTerm($term)) {
+        $this->addTermCornerGraphics($term, $details, $file_url_generator);
+      }
+    }
+
+    $this->list[0] = $this->createItem(0, $details);
+  }
+
+  /**
+   * Checks if the node has any corner graphic fields with values.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node entity.
+   *
+   * @return bool
+   *   TRUE if the node has corner graphics, FALSE otherwise.
+   */
+  protected function hasCornerGraphicsInNode(NodeInterface $node) {
+    return ($node->hasField('field_graphical_image') && !$node->field_graphical_image->isEmpty()) ||
+      ($node->hasField('field_bottom_graphical_image') && !$node->field_bottom_graphical_image->isEmpty());
+  }
+
+  /**
+   * Adds node corner graphics to the details array.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node entity.
+   * @param array $details
+   *   The details array to be modified.
+   * @param object $file_url_generator
+   *   The file URL generator service.
+   */
+  protected function addNodeCornerGraphics(NodeInterface $node, array &$details, $file_url_generator) {
+    if ($node->hasField('field_graphical_image') &&
```

Comment:
Where is field_graphical_image added to the node bundles?

Reply:
`tide_core` defines `field_graphical_image`, but whether to use it depends on the modules being enabled.